retrieve/retrieve.go

```diff
@@ -0,0 +1,94 @@
+package retrieve
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/joshuatcasey/libdependency/buildpack_config"
+	"github.com/joshuatcasey/libdependency/versionology"
+	"github.com/joshuatcasey/libdependency/workflows"
+	"github.com/paketo-buildpacks/packit/v2/fs"
+)
+
+// GenerateMetadataFunc is a function type that buildpack authors will implement and pass in to NewMetadata.
+// Given a versionology.VersionFetcher, the implementation must return the associated metadata for that version.
+// If there are multiple targets for the same version, return multiple versionology.Dependency.
+type GenerateMetadataFunc func(version versionology.VersionFetcher) ([]versionology.Dependency, error)
+
+// NewMetadata is the entrypoint for a buildpack to retrieve new versions and the metadata thereof.
+// Given a way to retrieve all versions (getNewVersions) and a way to generate metadata for a version (generateMetadata),
+// this function will take in the dependency workflow inputs and the dependency workflow outputs
+func NewMetadata(id string, getNewVersions buildpack_config.VersionFetcherFunc, generateMetadata GenerateMetadataFunc) {
+	buildpackTomlPath, output := FetchArgs()
+	validate(buildpackTomlPath, output)
+
+	config, err := buildpack_config.ParseBuildpackToml(buildpackTomlPath)
+	if err != nil {
+		panic(err)
```

Comment:
I am in agreement with Ryan's comment about the code panicking:
> It panics on errors. That doesn't really seem like an ideal API. What if I want to clean something up before the program exits, am I expected to recover?

I'd love to see this resolved.